python_modules/libraries/dagster-pyspark/dagster_pyspark/resources.py

```diff
@@ -28,6 +33,48 @@ def spark_context(self):
         return self.spark_session.sparkContext
 
 
+class PySparkResource(ConfigurableResource, IAttachDifferentObjectToOpContext):
+    """This resource provides access to a PySpark Session for executing PySpark code within Dagster.
+
+    Example:
+        .. code-block:: python
+
+            @op
+            def my_op(pyspark: PySparkResource)
+                spark_session = pyspark.get_client().spark_session
+                dataframe = spark_session.read.json("examples/src/main/resources/people.json")
+
+
+            @job(
+                resource_defs={
+                    "pyspark": PySparkResource(
+                        spark_config={
+                            "spark.executor.memory": "2g"
+                        }
+                    )
+                }
+            )
+            def my_spark_job():
+                my_op()
+    """
+
+    spark_config: Dict[str, Any]
+
+    @cached_method
+    def get_client(self) -> PySparkClient:
+        result = validate_config(spark_config(), self.spark_config)
+        if not result.success:
+            raise DagsterInvalidConfigError(
+                "Error when processing PySpark resource config ",
+                result.errors,
+                self.spark_config,
+            )
+        return PySparkClient(self.spark_config)
+
+    def get_object_to_set_on_execution_context(self) -> Any:
+        return self.get_client()
+
+
```

Comment:
should we be using lifecycle methods in lieu of IAttachDifferentObjectToOpContext at this point?

Reply:
Yup, updated.